packages/demos-babylon/src/chatbotDemo.js

```diff
@@ -0,0 +1,182 @@
+let transcriptTextEl;
+let lex;
+
+function initConversationManagement() {
+  // Use talk button events to start and stop recording.
+  const talkButton = document.getElementById('talkButton');
+  talkButton.onmousedown = () => lex.beginVoiceRecording();
+  talkButton.onmouseup = () => lex.endVoiceRecording();
+
+  // Use events dispatched by the LexFeature to present helpful user messages.
+  const { EVENTS } = AwsFeatures.LexFeature;
+  lex.listenTo(EVENTS.lexResponseReady, (response) => handleLexResponse(response));
+  lex.listenTo(EVENTS.recordBegin, () => hideUserMessages());
+  lex.listenTo(EVENTS.recordEnd, () => displayProcessingMessage());
+
+  // Create convenience references to DOM elements.
+  messageContainerEl = document.getElementById('userMessageContainer');
+  transcriptTextEl = document.getElementById('transcriptText');
+}
+
+/**
+ * Triggered whenever a response is received from the Lex chatbot.
+ * @param {object} response An object representing the Lex response. For a
+ * detailed description of this object's shape, see the documentation for the
+ * "data" callback argument described here:
+ * {@link https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/LexRuntime.html#postContent-property}
+ */
+function handleLexResponse(response) {
+  // Remove "processing" CSS class from message container.
+  messageContainerEl.classList.remove('processing');
+
+  // Display the user's speech input transcript.
+  displaySpeechInputTranscript(response.inputTranscript);
+
+  // Have the host speak the response from Lex.
+  host.TextToSpeechFeature.play(response.message);
+}
+
+function displaySpeechInputTranscript(text) {
+  transcriptTextEl.innerText = `“${text}”`;
+  messageContainerEl.classList.add('showingMessage');
+}
+
+function displayProcessingMessage() {
+  messageContainerEl.classList.add('processing');
+}
+
+function hideUserMessages() {
+  messageContainerEl.classList.remove('showingMessage');
+}
+
+/**
+ * Attempts to enable microphone access for Lex, triggering a browser permissions
+ * prompt if necessary.
+ */
+function acquireMicrophoneAccess() {
+  showUiScreen('micInitScreen');
+
+  lex.enableMicInput()
+    .then(() => {
+      // Microphone access is complete. Display the start screen.
+      showUiScreen('startScreen');
+    })
+    .catch((e) => {
+      // The user or browser denied mic access. Display appropriate messaging
+      // to the user.
+      switch (e.message) {
+        case 'Permission dismissed':
+          showUiScreen('micPermissionDismissedScreen');
+          break;
+        default:
+          showUiScreen('micDisabledScreen');
+          break;
+      }
+    });
+}
+
+// ===== Utility functions =====
+
+/**
+ * Makes the specified UI screen visible and hides all other UI screens.
+ * @param {string} id HTMLElement id of the screen to display.
+ */
+function showUiScreen(id) {
+  document.querySelectorAll('#uiScreens .screen').forEach((element) => {
+    const show = element.id === id;
```

Comment:
nit: `show` is not a a good name for a boolean variable.

Reply:
Can you tell me more? What makes "show" a poor choice? What characteristics would make for a better name?

Reply:
From my perspective, `isVisiable` is a better one. I also noticed in the set method, the argument name is `visible`.